### ISSUE TITLE
Support for New Architecture / Interop Layer

### ISSUE THREAD

**Issue by javascripter** [software-mansion-labs/react-native-turbo-demo#178]
### Description

I know it may be a bit early as react-native-turbo is still in beta, but since React Native 0.74 will enable the New Architecture by default with bridgeless mode on, I'd like to know if there is a plan for react-native-turbo to support the new Architecture directly or via Interop Layer.

- https://github.com/reactwg/react-native-new-architecture/discussions/167
- https://docs.expo.dev/guides/new-architecture/

So far I've tried an Expo 50 app with New Arch with the following `react-native.config.js`, but it does not seem to work.
```
module.exports = {
  project: {
    ios: {
      unstable_reactLegacyComponentNames: ['RNVisitableView'],
    },
  },
}
```

### Steps to reproduce

https://docs.expo.dev/guides/new-architecture/#enable-the-new-architecture-in-an-existing-project

Follow the steps in the docs to enable New Arch and run with the react-native-config.js above.

### Snack or a link to a repository

https://github.com/software-mansion-labs/react-native-turbo-demo/tree/main/examples/turbo-demo-expo-example

### React Native Turbo version

1.0.0-beta.5

### React Native version

0.73.6

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

Expo Dev Client

### Architecture

Fabric (New Architecture)

### Build type

Debug app & dev bundle

### Device

iOS simulator

**Comment by pklatka** [comment]
Hi @javascripter,

Initially, this was in the plan, but we've been focusing on making this library work as well as possible. The library will be out of beta soon (we're still testing it and not planning any major changes), so this probably should be done after we're out of beta.

**Comment by javascripter** [comment]
@pklatka
Makes sense! It's great to know New Arch will be considered at some point after a stable release too.
Thanks for working on this project. I believe this library has a lot of potential and enables writing universal apps with great UX & DX.
